src/resource.c

```diff
@@ -548,10 +548,9 @@ coap_free_resource(coap_resource_t *resource) {
 
 #ifdef WITH_LWIP
   memp_free(MEMP_COAP_RESOURCE, resource);
-#endif
-#ifndef WITH_LWIP
+#else
   coap_free_type(COAP_RESOURCE, resource);
-#endif /* WITH_CONTIKI */
+#endif
```

Comment:
This actually should be (as` coap_free_type(type, p)` is mapped to `memp_free(MEMP_ ## type, p)` for LwIP)
````
@@ -526,12 +526,7 @@ coap_free_resource(coap_resource_t *resource) {
     coap_free(resource->proxy_name_list);
   }
 
-#ifdef WITH_LWIP
-  memp_free(MEMP_COAP_RESOURCE, resource);
-#endif
-#ifndef WITH_LWIP
   coap_free_type(COAP_RESOURCE, resource);
-#endif /* WITH_CONTIKI */
 }
 
 void
````

Reply:
You have additionally found and fixed the` coap_free_type(COAP_RESOURCEATTR, attr);` - thanks, but not the `coap_free_type(COAP_RESOURCE, resource);` as suggested above.